flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/table/lookup/fullcache/LookupFullCache.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.runtime.functions.table.lookup.fullcache;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.metrics.groups.CacheMetricGroup;
+import org.apache.flink.table.connector.source.lookup.LookupOptions.LookupCacheType;
+import org.apache.flink.table.connector.source.lookup.cache.LookupCache;
+import org.apache.flink.table.connector.source.lookup.cache.trigger.CacheReloadTrigger;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.util.Preconditions;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/** Internal implementation of {@link LookupCache} for {@link LookupCacheType#FULL}. */
+public class LookupFullCache implements LookupCache {
+    private static final long serialVersionUID = 1L;
+
+    private final CacheLoader cacheLoader;
+    private final CacheReloadTrigger reloadTrigger;
+
+    private volatile boolean metricsOpened;
+    private transient volatile ReloadTriggerContext reloadTriggerContext;
+    private transient volatile Throwable reloadFailCause;
+
+    public LookupFullCache(CacheLoader cacheLoader, CacheReloadTrigger reloadTrigger) {
+        this.cacheLoader = Preconditions.checkNotNull(cacheLoader);
+        this.reloadTrigger = Preconditions.checkNotNull(reloadTrigger);
+    }
+
+    @Override
+    public synchronized void open(CacheMetricGroup metricGroup) {
+        if (!metricsOpened) {
+            cacheLoader.open(metricGroup);
+            metricsOpened = true;
+        }
+    }
+
+    public synchronized void open(Configuration parameters) throws Exception {
+        if (reloadTriggerContext == null) {
+            cacheLoader.open(parameters);
+            reloadTriggerContext =
+                    new ReloadTriggerContext(cacheLoader, th -> reloadFailCause = th);
```

Comment:
The exception consumer could be like storing the exception if `reloadFailCause` is null, otherwise `reloadFailCause.addSuppressed(th)`, to reveal all caught errors